core/src/main/scala/org/apache/spark/sql/rapids/tool/util/StringUtils.scala

```diff
@@ -56,6 +56,51 @@ object StringUtils extends Logging {
     "\"" + str.replace("\"", "\"\"") + "\""
   }
 
+  def escapeMetaCharacters(str: String): String = {
+    str.replaceAll("\n", "\\\\n")
+      .replaceAll("\r", "\\\\r")
+      .replaceAll("\t", "\\\\t")
+      .replaceAll("\f", "\\\\f")
+      .replaceAll("\b", "\\\\b")
+      .replaceAll("\u000B", "\\\\v")
+      .replaceAll("\u0007", "\\\\a")
+  }
+
+/**
+   * Process a string from Spark info objects. It is used as a wrapper to truncate the string.
```

Comment:
we can probably clarify this a bit ("Spark info objects" feels awkward) or make it more generic just saying formatting a string based on parameters